docs/tokenserver/tokenserver_api.md

```diff
@@ -0,0 +1,157 @@
+# Token Server API v1.0
+===============
+
+.. note::
+
+    Unless stated otherwise, all APIs are using application/json for the requests
+    and responses content types.
+
+
+**GET** **/1.0/<app_name>/<app_version>**
+
+    Asks for new token given some credentials in the Authorization header.
+
+    By default, the authentication scheme is Mozilla Accounts OAuth 2.0
+    but other schemes can
+    potentially be used if supported by the login server.
+
+    - **app_name** is the name of the application to access, like **sync**.
+    - **app_version** is the specific version number of the api that you want
+      to access.
+
+    The first /1.0/ in the URL defines the version of the authentication
+    token itself.
+
+    Example for Mozilla Account OAuth 2.0::
+
+        GET /1.0/sync/1.5
+        Host: token.services.mozilla.com
+        Authorization: bearer <assertion>
+
+    This API returns several values in a json mapping:
+
+    - **id** -- a signed authorization token, containing the
+      user's id for the application and the node.
+    - **key** -- a secret derived from the shared secret
+    - **uid** -- the user id for this service
+    - **api_endpoint** -- the root URL for the user for the service.
+    - **duration** -- the validity duration of the issued token, in seconds.
+
+    Example::
+
+        HTTP/1.1 200 OK
+        Content-Type: application/json
+
+        {'id': <token>,
+         'key': <derived-secret>,
+         'uid': 12345,
+         'api_endpoint': 'https://db42.sync.services.mozilla.com/1.5/12345',
+         'duration': 300,
+        }
+
+    If the **X-Client-State** header is included in the request, the
+    server will compare the submitted value to any previously-seen value.
+    If it has changed then a new uid and api_endpoint are generated, in
+    effect "resetting" the node allocation for this user.
+
+
+## Request Headers
+===============
+
+
+**X-Client-State**
+
+    An optional string that can be sent to identify a unique configuration
+    of client-side state.  It may be up to 32 characters long, and must
+    contain only characters from the urlsafe-base64 alphabet (i.e.
+    alphanumeric characters, underscore and hyphen) and the period.
+
+    A change in the value of this header will cause the user's node
```

Comment:
Well, _may_ cause the user's node allocation to be reset. 

Sync currently has only one node.